Cc: James E.J. Bottomley <James.Bottomley@SteelEye.com>,
    linux-scsi@vger.kernel.org
Subject: [PATCH] m68k: Atari SCSI workqueue updates

Workqueue updates for the Atari SCSI driver

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
---
 drivers/scsi/atari_NCR5380.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- linux-m68k-2.6.21.orig/drivers/scsi/atari_NCR5380.c
+++ linux-m68k-2.6.21/drivers/scsi/atari_NCR5380.c
@@ -657,7 +657,7 @@ static inline void NCR5380_print_phase(s
 #include <linux/interrupt.h>
 
 static volatile int main_running;
-static DECLARE_WORK(NCR5380_tqueue, (void (*)(void *))NCR5380_main, NULL);
+static DECLARE_WORK(NCR5380_tqueue, NCR5380_main);
 
 static inline void queue_main(void)
 {
@@ -1075,7 +1075,7 @@ static int NCR5380_queue_command(Scsi_Cm
  *  reenable them.  This prevents reentrancy and kernel stack overflow.
  */
 
-static void NCR5380_main(void *bl)
+static void NCR5380_main(struct work_struct *work)
 {
 	Scsi_Cmnd *tmp, *prev;
 	struct Scsi_Host *instance = first_instance;
